src/interfaces.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{ActivityCreatorId, ActivityId};
+use near_contract_standards::non_fungible_token::metadata::{NFTContractMetadata, TokenMetadata};
+use near_sdk::AccountId;
+
+pub trait OwnerAction {
+    fn set_contract_metadata(&mut self, metadata: NFTContractMetadata) -> bool;
+
+    fn add_creator_whitelist(&mut self, account_ids: Vec<AccountId>);
+
+    fn remove_creator_whitelist(&mut self, account_ids: Vec<AccountId>);
+}
+
+pub trait ActivityCreatorAction {
+    fn create_activity_token_metadata(&mut self, token_metadata: TokenMetadata) -> ActivityId;
+
+    fn update_activity_token_metadata(
+        &mut self,
+        activity_id: ActivityId,
+        token_metadata: TokenMetadata,
+    ) -> TokenMetadata;
+
+    fn nft_mint(&mut self, activity_id: u32, nft_owner_ids: Vec<AccountId>);
+}
+
+pub trait View {
```

Comment:
Maybe we should add a view function to return the count of activities and/or the list of activity ids?